### Paso 4: Síntesis – Gini y Entropía sobre la misma variable (`estado_civil_id`)

Al aplicar ambos indicadores sobre la variable `estado_civil_id`, se obtiene una perspectiva dual y complementaria del fenómeno: mientras que la **entropía de Shannon** mide la uniformidad para describir qué tan repartida está la población entre los distintos estados civiles (apuntando a la heterogeneidad), el **coeficiente de Gini** aplicado a esas mismas frecuencias cuantifica la desigualdad o concentración relativa, permitiendo evaluar si un grupo reducido de categorías acapara la mayor parte de las observaciones frente a las demás.

In [5]:
from pathlib import Path
import sys

PROYECTO = Path.cwd().resolve()
if PROYECTO.name == 'notebooks':
    PROYECTO = PROYECTO.parent
if str(PROYECTO) not in sys.path:
    sys.path.insert(0, str(PROYECTO))

import polars as pl
import polars.selectors as cs
from src.analysis.gini_vs_entropia import (
    cargar_datos,comparar_entropia_y_gini_categorias, VARIABLE

)

df = cargar_datos()
resultado_comparacion = comparar_entropia_y_gini_categorias(df,VARIABLE)
for k, v in resultado_comparacion.items():
    print(f"  - {k}: {v}")
print(f'Polars {pl.__version__}; registros: {df.height:,}')

  - variable: estado_civil_id
  - k_categorias: 6
  - Entropia_Shannon: 2.2466
  - Gini_Frecuencias: 0.3717
Polars 1.44.2; registros: 105,715


### Resultados e Interpretación para `estado_civil_id`

Al aplicar ambos indicadores sobre las frecuencias de las $6$ categorías de la variable `estado_civil_id`, se obtuvieron los siguientes valores:

* **Número de categorías ($K$):** $6$
* **Entropía de Shannon:** $2.2466$
* **Coeficiente de Gini (sobre frecuencias):** $0.3717$

#### Interpretación de los resultados:

1. **Entropía de Shannon ($2.2466$):** 
   Este valor es alto (considerando que el máximo teórico para $6$ categorías es $\approx 2.585$). Nos indica que hay una **gran uniformidad y dispersión** en la muestra; es decir, la población se encuentra ampliamente repartida entre los diferentes estados civiles en lugar de concentrarse en uno solo.

2. **Coeficiente de Gini aplicado a las frecuencias ($0.3717$):** 
   Este valor moderado-bajo de Gini complementa la lectura anterior. Al medir la concentración relativa entre las frecuencias de las categorías, un Gini de $0.3717$ confirma que **no existe una desigualdad extrema o un acaparamiento absoluto** por parte de una sola categoría. Aunque algunos estados civiles pueden tener más registros que otros, la distribución general se mantiene relativamente equilibrada y sin picos dominantes extremos.